_minutes/2024-11-07-wecg.md

```diff
@@ -0,0 +1,184 @@
+   * [oliver] I can follow up and leave a comment on the issue.
+ * [simeon] I'd suggest using 0 over undefined. It's falsey, it's a valid value for new Date(), it continues having the property be defined on the object. Most consistent with how this value may be used in the ecosystem.
+ * [timothy] 0, NaN?
+
+[Issue 715](https://github.com/w3c/webextensions/issues/715): Feature request: Enhance the tab group api to support the new pinning functionality
+
+ * [simeon] Feature request on tabGroups API to pin groups to the bookmarks.
+ * [oliver] I am not familiar with this.
+ * [rob] Does Safari support the tabGroups API?
+ * [timothy] We don't, but we have the feature and are interested in shoehorning the functionality in the API.
+ * [rob] In Firefox there's also interest in tab groups but we don't have the API yet.
+ * [oliver] The two requests that would be API level changes is to add a pinned property (Boolean) and tabs.query() include closed tab groups, which I'm not familiar with.
+ * [timothy] We have closed tab groups as well.
+ * [simeon] Would it make sense to have closed groups be a distinct type of tab to query?
+ * [timothy] Not necessarily.
+ * [rob] How does this functionality relate to sessions.restore? Based on the description it sounds similar.
+ * [oliver] Pinned fields sounds non-controversial, querying/managing closed groups sounds like something needing more discussion.
+
+[Issue 716](https://github.com/w3c/webextensions/issues/716): Inconsistent support for [context]menus.getTargetElement()
+
+ * [rob] Feature request for the ability to retrieve the element that is clicked when a context menu was opened. Firefox solved this by introducing getTargetElement(). When contextMenus.onClicked fires you can get a reference (targetElementId) that you pass to the menus.getTargetElement function in a content script to retrieve the element. There are workarounds such as exposing the coordinates in the document, but they're lossy. Firefox's approach is a reliable solution.
+ * [timothy] Against coordinate suggestion as well; interested in implementing Firefox's getTargetElement API.
+ * [oliver] Treating this as opaque is fine, but I'm curious what this ID looks like in Firefox.
+ * [rob] It's generated in Firefox. I believe it's the timestamp. On click, a weak ref to the element is stored, timestamp is captured and passed to the extension.
+ * [oliver] Thanks, just wanted to understand the general shape.
+ * [timothy] I'd want it to be opaque. UUID or some string.
+ * [rob] It's currently an integer. This enables extensions to pass the value to getTargetElement in a code string (tabs.executeScript) without serialization concerns. That concern is lesser in MV3 because scripting.executeScript does not accept code strings, and accepts args for func instead.
```

Comment:
```suggestion
 * [rob] It's currently an integer. This enables extensions to pass the value to `getTargetElement` in a code string (`tabs.executeScript`) without serialization concerns. That concern is lesser in MV3 because `scripting.executeScript` does not accept code strings, and accepts `args` for `func` instead.
```